Source/NSUnit.m

```diff
@@ -0,0 +1,1770 @@
+    {
+      [coder encodeDouble: _coefficient forKey: @"coefficient"];
+      [coder encodeDouble: _constant forKey: @"constant"];
+    }
+  else
+    {
+      [coder encodeValueOfObjCType: @encode(double) at: &_coefficient];
+      [coder encodeValueOfObjCType: @encode(double) at: &_constant];
+    }
+}
+
+- (double) coefficient
+{
+  return _coefficient;
+}
+
+- (double) constant
+{
+  return _constant;
+}
+
+- (double)baseUnitValueFromValue:(double)value
+{
+  return ((_coefficient * value) + _constant); 
+}
+
+- (double)valueFromBaseUnitValue:(double)baseUnitValue
+{
+  return ((baseUnitValue / _coefficient) - _constant);
+}
+@end
+
+// Abstract unit...
+@implementation NSUnit
++ (instancetype)new
+{
+  return [[self alloc] init];
+}
+           
+- (instancetype)init
+{
+  self = [super init];
+  if(self != nil)
+    {
+      ASSIGNCOPY(_symbol, @"");
+    }
+  return self;
+}
+          
+- (instancetype)initWithSymbol:(NSString *)symbol
+{
+  self = [super init];
+  if(self != nil)
+    {
+      ASSIGNCOPY(_symbol, symbol);
+    }
+  return self;
+}
+
+- (id) initWithCoder: (NSCoder *)coder
+{
+  if([coder allowsKeyedCoding])
+    {
+      _symbol = [coder decodeObjectForKey: @"symbol"];
+    }
+  else
+    {
+      _symbol = [coder decodeObject];
+    }
+  return self;
+}
+
+- (void) encodeWithCoder: (NSCoder *)coder
+{
+  if([coder allowsKeyedCoding])
+    {
+      [coder encodeObject: _symbol forKey: @"coder"];
+    }
+  else
+    {
+      [coder encodeObject: _symbol];
+    }
+}
+
+- (instancetype) copyWithZone: (NSZone *)zone
+{
+  NSUnit *u = [[NSUnit allocWithZone: zone] initWithSymbol: [self symbol]];
+  return u;
+}
+
+- (NSString *)symbol
+{
+  return _symbol;
+}
+@end
+
+
+// Dimension using units....
+@implementation NSDimension
+- (NSUnitConverter *) converter
+{
+  return _converter;
+}
+
+- (instancetype) initWithSymbol: (NSString *)symbol converter: (NSUnitConverter *) converter
+{
+  self = [super initWithSymbol: symbol];
+  if(self != nil)
+    {
+      ASSIGN(_converter, converter);
+    }
+  return self;
+}
+
+- (instancetype) initWithSymbol: (NSString *)symbol
+                    coefficient: (double)coefficient
+                       constant: (double)constant
+{
+  NSUnitConverterLinear *converter = [[NSUnitConverterLinear alloc] initWithCoefficient: coefficient
+                                                                               constant: constant];
+  NSDimension *result = [[[self class] alloc] initWithSymbol: symbol
+                                                   converter: converter];
+  return result;
+}
+
+
++ (instancetype) baseUnit
+{
+  return nil;
+}
+@end
+
+
+// Predefined....
+@implementation NSUnitAcceleration
+
++ (instancetype) baseUnit
+{
+  return [self metersPerSecondSquared];
+}
+
+// Base unit - metersPerSecondSquared
++ (NSUnitAcceleration *) metersPerSecondSquared
+{
+  NSUnitAcceleration *result = [[NSUnitAcceleration alloc] initWithSymbol: @"m/s^2"
```

Comment:
Shouldn't the results of all these methods be autoreleased?

Reply:
Ah, good point, yes.

Reply:
Done

Reply:
Sorry, I do not see the autorelease. Neither here nor in any of the other methods.